packages/flutter/test/material/back_button_test.dart

```diff
@@ -123,4 +122,23 @@ void main() {
     ));
     handle.dispose();
   });
+
+  testWidgets('CloseButton color test', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: Material(
+          child: CloseButton(
+            color: Colors.red,
+          ),
+        ),
+      ),
+    );
+
+    expect(
+      find.byWidgetPredicate(
+        (Widget widget) => widget is CloseButton && widget.color == Colors.red,
+      ),
+      findsOneWidget,
+    );
```

Comment:
While this does test that the `CloseButton` color param is getting set, it does it is kind of buried in the predicate. A more explicit way to do this might be:

```dart
    final CloseButton button = tester.firstWidget(find.byType(CloseButton));
    expect(button.color, Colors.red);
```

In addition, this is just testing if the color param is on the `CloseButton`, and not that the underlying icon is getting painted in that color.

As the icon is implemented using a specific character from a special icon font in a `RichText` widget, a better test might be to do something like:

```dart
    final RichText iconText = tester.firstWidget(find.descendant(
      of: find.byType(CloseButton),
      matching: find.byType(RichText)
    ));
    expect(iconText.text.style.color, Colors.red);
```

This will verify that the underlying widget will get the right color.

Reply:
Added both checks to the test